packages/ui/components/dialogs/loop/index.tsx

```diff
@@ -319,13 +342,117 @@ export default function Loop({
 
       currentTransactionStep++;
 
-      const tx = await currentSdk.createAndFundPositionAtRatio(
-        selectedCollateralAsset.cToken,
-        selectedBorrowAsset?.cToken ?? ('' as Address),
-        selectedCollateralAsset.underlyingToken,
-        amountAsBInt,
-        parseEther(currentLeverage.toString())
-      );
+      const previewDeposit = await publicClient.readContract({
+        abi: icErc20Abi,
+        address: selectedCollateralAsset.cToken,
+        functionName: 'previewDeposit',
+        args: [amountAsBInt]
+      });
+
+      const actualRedeemedAmountForAggregatorSwap =
+        await publicClient.readContract({
+          abi: icErc20Abi,
+          address: selectedCollateralAsset.cToken,
+          functionName: 'previewRedeem',
+          args: [previewDeposit]
+        });
+
+      // get initial quote to calculate slippage
+      const [, initialBorrowAmount] = await publicClient.readContract({
```

Comment:
when the ratio is decreased, the swap amount is derived from the first output variable `supplyAmount`, because the redeemed collateral should be exchanged for the other asset in order to repay the flash loan